extensions/wikia/Recirculation/js/recirculation.js

```diff
@@ -63,26 +65,94 @@ require([
 				height: 337,
 				modelName: 'wiki',
 				opts: {
-					rule_language : window.wgContentLanguage
+					rule_language: window.wgContentLanguage
 				}
 			}
 		};
 
 	function prepareRailRecirculation(options) {
+		var request;
+		var isRecirculationABTest = window.Wikia.AbTest.inGroup('RIGHT_RAIL_RECIRCULATION_SOURCE', 'TOPIC_FEED') &&
+			getCurationCMSTopic();
+
+		if (isRecirculationABTest) {
+			request = getDataFromCurationCMS();
+		} else {
+			request = liftigniter.prepare(options);
+		}
 		// prepare & render right rail recirculation module
-		liftigniter.prepare(options).done(function (data) {
+		request.done(function (data) {
 			require(['ext.wikia.recirculation.views.premiumRail'], function (viewFactory) {
 				var view = viewFactory();
 				view.render(data, options.title)
 					.then(view.setupTracking())
 					.then(function () {
-						liftigniter.setupTracking(view.itemsSelector, options);
+						if (!isRecirculationABTest) {
+							liftigniter.setupTracking(view.itemsSelector, options);
+						}
 					});
 			});
 		});
 	}
 
-	function prepareEnglishRecirculation () {
+	function getNormalizedCurationCMSData(data) {
+		var normalizedData = {};
+		normalizedData.items = [];
+
+		data.posts.forEach(function (post) {
+			normalizedData.items.push({
+				title: post.title,
+				thumbnail: post.thumbnail.url,
+				url: post.url
+			});
+		});
+
+		return normalizedData;
+	}
+
+	function getDataFromCurationCMS() {
+		var deferred = $.Deferred();
+
+		nirvana.sendRequest({
+			controller: 'RecirculationApi',
+			method: 'getFandomPosts',
+			format: 'json',
+			scriptPath: window.wgCdnApiUrl,
+			data: {
+				type: 'stories',
+				slug: getCurationCMSTopic()
+			},
+			callback: function (data) {
```

Comment:
afair, `sendRequest` returns `$.ajax` which allows you to call `nirvana.sendRequest().done().fail()`